### PR TITLE
Benchmarks for lightclient pallet

### PR DESCRIPTION
**Pull request by Rizziepit**
This PR implements worst case and average case benchmarks for the Eth lightclient pallet. The absolute worst case is used to set the extrinsic weight. In a follow-up PR, we should use the average cases to "refund" weight.

Variability in perf will mainly come from:
* pruning more / fewer headers and whether those headers are siblings
* finalizing a new header
* early fail during validation - I am not planning on bothering with a refund here

NOTE: if we want to make the pallet more generic, we could make `DescendantsUntilFinalized` a benchmark parameter. For now, I'm benchmarking for a single runtime only.

### CODE REVIEW

File: parachain/pallets/verifier-lightclient/Cargo.toml [hunk 1]
```diff
@@ -49,3 +51,9 @@ std = [
     "artemis-core/std",
     "artemis-ethereum/std",
 ]
+runtime-benchmarks = [
+    "frame-benchmarking",
+    "frame-support/runtime-benchmarks",
+    "frame-system/runtime-benchmarks",
+    "hex-literal",
+]
```
Comment (vgeddes) [inline parachain/pallets/verifier-lightclient/Cargo.toml:59]: newline! 😛

**Review by vgeddes — APPROVED** [review]
Looks great, thanks Riz! 🚀 

Can merge, once you've sorted out the failing Github Actions check:
```
31 | impl<T: frame_system::Config> verifier_lightclient::WeightInfo for WeightInfo<T> {
   |                               ^^^^^^^^^^^^^^^^^^^^ use of undeclared crate or module `verifier_lightclient`
```

**Comment by Rizziepit** [comment]
> Looks great, thanks Riz! rocket
> 
> Can merge, once you've sorted out the failing Github Actions check:
> 
> ```
> 31 | impl<T: frame_system::Config> verifier_lightclient::WeightInfo for WeightInfo<T> {
>    |                               ^^^^^^^^^^^^^^^^^^^^ use of undeclared crate or module `verifier_lightclient`
> ```

Lol. Yes. I should fix this haha

**Review by philipstanislaus — COMMENTED** [review]
Nice 🎉 

Sorry for late review. What hardware did you use to run those benchmarks?

**Comment by Rizziepit** [comment]
> Sorry for late review. What hardware did you use to run those benchmarks?

I ran them on my machine (3.6 GHz CPU, 32 GB memory, NVMe SSD). The plan is to re-run them on a compute-optimized AWS instance with SSD (c5d.xlarge?). https://github.com/Snowfork/polkadot-ethereum/blob/main/parachain/scripts/benchmark.sh automates the whole process

**Comment by philipstanislaus** [comment]
@Rizziepit cool, I have added a note to this open ticket: https://github.com/Snowfork/polkadot-ethereum/issues/161 for re-running them with hardware comparable to what Parity did.

**Pull request merged**